src/crimson/common/logfile.cc

```diff
@@ -0,0 +1,73 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <fstream>
+#include <seastar/util/log.hh>
+
+#include "common/debug.h"
+#include "logfile.h"
+#include "log.h"
+
+#define dout_context g_ceph_context
+#define dout_prefix *_dout
+#define dout_subsys ceph_subsys_
+
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_osd);
+  }
+}  // namespace
+
+struct logfh {
+  std::string log_file_path;
+  std::ofstream log_ofstream;
+
+  ~logfh() {
+    log_ofstream.flush();
+    log_ofstream.close();
+  }
+  
+  bool is_open() const {
+    return !log_file_path.empty() && log_ofstream.is_open();
+  }
+  int open(std::string_view log_file);
+};
+
+std::unique_ptr<logfh> lfh;
```

Comment:
I would add a new line here. At first glance it looked like a return type of `logfh::open`.